### PR TITLE
Stop appending `.nii.gz` to the centerline output filepath if `-o` has an extension

### LINKED ISSUE #3530: Additional extension in output file
### Description

When running something like this:
~~~
 sct_get_centerline -i sag_cervical_seg.nii.gz -method fitseg -o sag_cervical_centerline.nii.gz
~~~

I expect output file to be `sag_cervical_centerline.nii.gz`, but instead it is `sag_cervical_centerline.nii.gz.nii.gz`

### PR DESCRIPTION
**Pull request by joshuacwnewton**
#### GitHub

- [ ] I've given this PR a concise, self-descriptive, and meaningful title
- [ ] I've linked relevant issues in the PR body
- [ ] I've applied [the relevant labels](https://intranet.neuro.polymtl.ca/geek-tips/contributing#pr-labels-a-href-pr-labels-id-pr-labels-a) to this PR
- [ ] I've applied a [release milestone](https://github.com/spinalcordtoolbox/spinalcordtoolbox/milestones) (major, minor, patch) in line with [Semantic Versioning guidelines](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Misc%3A-Creating-a-new-release#convention-for-naming-releases) 
- [ ] I've assigned a reviewer

#### PR contents

- [ ] I've consulted [SCT's internal developer documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

This PR adds a failing test to reproduce the behavior in #3530, then changes the behavior of the script to stop appending `.nii.gz` (which makes the test pass).

## Linked issues

Fixes #3530.

### CODE REVIEW

File: spinalcordtoolbox/scripts/sct_get_centerline.py [hunk 1]
```diff
@@ -172,20 +172,20 @@ def main(argv=None):
...
     qc_dataset = arguments.qc_dataset
     qc_subject = arguments.qc_subject
 
     # Generate QC report
     if path_qc is not None:
-        generate_qc(fname_input_data, fname_seg=file_output + '.nii.gz', args=sys.argv[1:], path_qc=os.path.abspath(path_qc),
+        generate_qc(fname_input_data, fname_seg=file_output, args=sys.argv[1:], path_qc=os.path.abspath(path_qc),
                     dataset=qc_dataset, subject=qc_subject, process='sct_get_centerline')
-    display_viewer_syntax([fname_input_data, file_output + '.nii.gz'], colormaps=['gray', 'red'], opacities=['', '0.7'])
+    display_viewer_syntax([fname_input_data, file_output], colormaps=['gray', 'red'], opacities=['', '0.7'])
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/scripts/sct_get_centerline.py:188]: Looking at the changes again, I'm realizing that we're actually duplicating our call to `display_viewer_syntax` unnecessarily. 

Now is probably a good time to remove this, aha. :sweat_smile:

Comment (joshuacwnewton) [reply spinalcordtoolbox/scripts/sct_get_centerline.py]: Done in a6c8728.

**Review by RignonNoel — COMMENTED** [review]
LGTM, however it change the interface of the script, so I don't know if it's really what you want since it's a breaking change.

For me there is two option:

1. Allow people to put it with OR without extension to support the old interface (we could use the ext of the input file by default if there is none)
2. Follow a deprecation process (Idk if we have one ?)

**Comment by RignonNoel** [comment]
> We actually just want to make sure we always output compressed NiFTI files, even if the input data file is `.nii`.

@joshuacwnewton But what about the case where the user give a `-o myfile.nii` ? Actually you're creating a `.nii` and not a `.nii` even if your output is comrpessed.

Should we not just remove the extension of the `-o` if there is one and export always as a `.nii.gz` ?

Or... We do a breaking change by adding an error message if somebody give an ext in the `-o` param

**Comment by joshuacwnewton** [comment]
Apologies, my commit message was unclear! :sweat_smile: 

> We actually just want to make sure we always output compressed NiFTI files, even if the input data file is `.nii`.

What I should have said was:

* If the user doesn't specify the output extension (either by not providing `-o`, or by providing `-o filename`), then default to `.nii.gz`.
* If the user _does_ specify the output extension (either by `filename.nii` or `filename.nii.gz`), then we should respect their choice. (Just in case the user wants uncompressed NIfTI files.)

**Comment by RignonNoel** [comment]
Ok, in this case we are on the same page and everything seems good for me.

**Pull request merged**